tutoraspects/templates/aspects/apps/superset/pythonpath/create_assets.py

```diff
@@ -61,26 +62,45 @@ def main():
 def create_assets():
     """Create assets from a yaml file."""
     roles = {}
+
+    for root, dirs, files in os.walk(ASSETS_PATH):
+        for file in files:
+            if not file.endswith(".yaml"):
+                continue
+
+            path = os.path.join(root, file)
+            with open(path, "r") as file:
+                asset = yaml.safe_load(file)
+                if not asset:
+                    continue
+
+                # Process the asset directly
+                if FILE_NAME_ATTRIBUTE in asset:
+                    file_name = asset.pop(FILE_NAME_ATTRIBUTE)
+
+                    # Find the right folder to create the asset in
+                    for asset_name, folder in ASSET_FOLDER_MAPPING.items():
+                        if asset_name in asset:
+                            write_asset_to_file(asset, asset_name, folder, file_name, roles)
+                            break
+
     with open(ASSETS_FILE_PATH, "r") as file:
         extra_assets = yaml.safe_load(file)
 
-        if not extra_assets:
-            print("No extra assets to create")
-            return
+        if extra_assets:
+            # For each asset, create a file in the right folder
+            for asset in extra_assets:
+                if FILE_NAME_ATTRIBUTE not in asset:
```

Comment:
Same comment as above. You can restore the condition and use a break statement

Reply:
`break` or `continue` statement do not work for non loop statements. I have being searching and not appear have a way to skip only the `with` statement. Any ideas?

Reply:
Leave it as it's